### PR TITLE
Ported ServerIntegrationTest to DefaultGNSTest

### PR DESCRIPTION
**Pull request by v-arun**

### CODE REVIEW

**Comment by westyUMass** [comment]
There are some conflicts because of the merge done last night that need to be resolved before it can be merged. Should I fix them? It appears I can't use the web editor.

File: test/edu/umass/cs/gnsclient/client/integrationtests/ServerIntegrationTest.java [hunk 1]
```diff
@@ -15,100 +15,85 @@
...
-import edu.umass.cs.gnsclient.jsonassert.JSONAssert;
-import edu.umass.cs.gnsclient.jsonassert.JSONCompareMode;
-import edu.umass.cs.gnscommon.utils.Base64;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.not;
 
-import java.io.BufferedReader;
+import java.awt.geom.Point2D;
```
Comment (akarthik10) [inline test/edu/umass/cs/gnsclient/client/integrationtests/ServerIntegrationTest.java]: `java.awt`?

Comment (westyUMass) [inline test/edu/umass/cs/gnsclient/client/integrationtests/ServerIntegrationTest.java]: Seems like that should be flagged by the Eclipse warnings as no longer being used.

**Comment by akarthik10** [comment]
Additional points to note:
1. There are way too many exceptions in Travis logs. Is this expected?
Eg:
```
java.lang.AssertionError
    [junit] 	at edu.umass.cs.gigapaxos.PaxosManager.unpause(PaxosManager.java:2330)
    [junit] 	at edu.umass.cs.gigapaxos.PaxosManager.getInstance(PaxosManager.java:1755)
    [junit] 	at edu.umass.cs.gigapaxos.PaxosManager.getInstance(PaxosManager.java:1764)
    [junit] 	at edu.umass.cs.gigapaxos.PaxosManager.handlePaxosPacket(PaxosManager.java:1111)
    [junit] 	at edu.umass.cs.gigapaxos.PaxosManager.enqueueRequest(PaxosManager.java:1073)
    [junit] 	at edu.umass.cs.gigapaxos.PaxosManager.handleIncomingPacket(PaxosManager.java:1049)
    [junit] 	at edu.umass.cs.gigapaxos.PaxosManager.access$7(PaxosManager.java:1039)
    [junit] 	at edu.umass.cs.gigapaxos.PaxosManager$FastDemultiplexer.handleMessage(PaxosManager.java:959)
    [junit] 	at edu.umass.cs.nio.AbstractPacketDemultiplexer$Tasker.run(AbstractPacketDemultiplexer.java:377)
    [junit] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    [junit] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    [junit] 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
    [junit] 	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
    [junit] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    [junit] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    [junit] 	at java.lang.Thread.run(Thread.java:745)
    [junit] killing RC3.3(29872) RC3.2(29845) RC3.1(29819) GNSApp3.3(29800) GNSApp3.2(29781) GNSApp3.1(29765)
```

```
 [junit] edu.umass.cs.gnscommon.exceptions.client.InvalidGuidException: BAD_GUID_ERROR: +NO+ +BADGUID+ 22D3647FB1693D3E1B803F4828D5A489BDC5145D for command COMMAND:RemoveAccount:440:22D3647FB1693D3E1B803F4828D5A489BDC5145D:4964464116261859328
    [junit] 	at edu.umass.cs.gnsclient.client.CommandUtils.checkResponse(CommandUtils.java:448)
    [junit] 	at edu.umass.cs.gnsclient.client.GNSClient.sendSync(GNSClient.java:307)
    [junit] 	at edu.umass.cs.gnsclient.client.GNSClient.sendSync(GNSClient.java:405)
    [junit] 	at edu.umass.cs.gnsclient.client.GNSClient.sendSync(GNSClient.java:417)
    [junit] 	at edu.umass.cs.gnsclient.client.GNSClient.execute(GNSClient.java:647)
    [junit] 	at edu.umass.cs.gnsserver.utils.DefaultGNSTest.removeCreatedState(DefaultGNSTest.java:343)
    [junit] 	at edu.umass.cs.gnsserver.utils.DefaultGNSTest.tearDownAfterClass(DefaultGNSTest.java:329)
    [junit] 	at edu.umass.cs.gnsserver.utils.DefaultGNSTest.tearDownAfterClass(DefaultGNSTest.java:321)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    [junit] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    [junit] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    [junit] 	at java.lang.reflect.Method.invoke(Method.java:498)
    [junit] 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    [junit] 	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    [junit] 	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    [junit] 	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
    [junit] 	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
    [junit] 	at junit.framework.JUnit4TestAdapter.run(JUnit4TestAdapter.java:38)
    [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.run(JUnitTestRunner.java:518)
    [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.launch(JUnitTestRunner.java:1052)
    [junit] 	at org.apache.tools.ant.taskdefs.optional.junit.JUnitTestRunner.main(JUnitTestRunner.java:906)
```
2. The CI test time has increased from 22 mins to 37 mins.

File: test/edu/umass/cs/gnsclient/client/integrationtests/ServerIntegrationTest.java (hunk 1 above)
Comment (akarthik10) [inline test/edu/umass/cs/gnsclient/client/integrationtests/ServerIntegrationTest.java]: I realized that `java.awt.geom.Point2D` is really being used. I take back this comment.

**Comment by v-arun** [comment]
Who is prhero and why do they heart me at the top? :)

**Comment by v-arun** [comment]
FYI, my IDE settings don't show a warning for public test methods that are being overridden because the parent is expected to have a javadoc. I don't see travis_checks flagging this as a warning either.

**Comment by akarthik10** [comment]
> Who is prhero and why do they heart me at the top? :)

@prhero https://github.com/njbmartin/prhero

**Comment by v-arun** [comment]
Westy, please merge these conflicts. The previous failures were apparently because my merge in 350abca was not okay, so you should do it this time.

**Comment by westyUMass** [comment]
Dealing with some issues with the app currently. I should get to this this afternoon.

**Comment by westyUMass** [comment]
Hmmm.. a bit of a holdup. I followed these steps suggested to begin to resolve the conflicts via the command line:

git fetch origin
git checkout -b arun origin/arun
git merge master

But what I get after running that is that my arun branch is missing some updates that are in the master branch. Odd. Still digging.

**Comment by westyUMass** [comment]
The master has newer commits to src/edu/umass/cs/gnsserver/gnsapp/clientCommandProcessor/commandSupport/AccountAccess.java 
that don't appear when I "git merge master" to bring them in to the arun branch. Not sure how to resolve this.

**Comment by v-arun** [comment]
What is an example of something that doesn’t appear? I see AccountAccess being auto-merged without a conflict.

**Comment by westyUMass** [comment]
Yes, AccountAccess is being merged without conflict but the resulting version is from your branch and not the master. Your branch contains an older version of that file. The same is true in many other places. It's like git thinks all your changes to those files (which I don't believe you actually changed) are more recent than the ones in the master.

**Comment by v-arun** [comment]
Maybe it’s because of the merge revert I did. I’ll force revert to 97a9785e8c5c2d460e717ac4c31590d2339cca2e
 and redo my simple changes since then.

**Comment by westyUMass** [comment]
Cool. Also check out the rebase stuff. It might make your branch merges simpler in the future. Maybe.

**Comment by v-arun** [comment]
Sorry that isn’t as easy as I thought. If I revert to 97a9785e8c5c2d460e717ac4c31590d2339cca2e I am losing TestWatcher I added in DefaultGNSTest that seems to be intertwined with the merge 2a675fc7c5a558f9e50b2a54158d583f8130f694. This needs some careful manual comparison, so I need to pass the ball back to you or a greater git expert.

**Comment by v-arun** [comment]
Try merging now.

**Comment by westyUMass** [comment]
OK. This looks better. I'm working on the merge now.

**Pull request merged**